### PR TITLE
DrivAerML: higher LR (7e-4, 8e-4, 1e-3) + gc=0.5 compound

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Three experiments (PRs #2797 ray, #2779 levi, #2781 chrome-previous) confirmed lower LR (3e-4) hurts DrivAerML — the model needs at least lr=5e-4 to converge within the 180-min budget. The question is whether HIGHER LR helps. On AirfRANS, gc=0.5 allowed more aggressive optimization steps, yielding 28.9% improvement. DrivAerML may have a similar dynamic: higher LR + gc=0.5 stability could reach deeper minima faster. This tests lr=7e-4, 8e-4, and 1e-3, all combined with gc=0.5 to prevent the gradient spikes that would destabilize higher LRs. Note: shinobu (#2806) has lr=7e-4 WITHOUT gc — this provides the gc comparison.

## Instructions

Run 4 experiments in parallel (2 GPUs each):

**Run 1 — lr=7e-4 + gc=0.5**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0,1 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset drivaerml --optimizer adamw \
  --lr 7e-4 --cosine-t-max 30 --grad-clip 0.5 \
  --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 \
  --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group drivaerml-lr-gc \
  --wandb-name driv-lr7e4-gc05
```

**Run 2 — lr=8e-4 + gc=0.5**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=2,3 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset drivaerml --optimizer adamw \
  --lr 8e-4 --cosine-t-max 30 --grad-clip 0.5 \
  --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 \
  --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group drivaerml-lr-gc \
  --wandb-name driv-lr8e4-gc05
```

**Run 3 — lr=1e-3 + gc=0.5**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=4,5 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset drivaerml --optimizer adamw \
  --lr 1e-3 --cosine-t-max 30 --grad-clip 0.5 \
  --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 \
  --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group drivaerml-lr-gc \
  --wandb-name driv-lr1e3-gc05
```

**Run 4 — lr=5e-4 + gc=0.5 + WD=1e-2 (compound regularization)**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=6,7 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset drivaerml --optimizer adamw \
  --lr 5e-4 --cosine-t-max 30 --grad-clip 0.5 --weight-decay 1e-2 \
  --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 \
  --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group drivaerml-lr-gc \
  --wandb-name driv-lr5e4-gc05-wd1e2
```

Report best val_primary/surface_rel_l2_pct checkpoint for each run. Compare all against baseline 4.619% and note whether gc=0.5 + higher LR outperforms shinobu's lr=7e-4 baseline (no gc).

## Baseline

- **Primary metric:** val_primary/surface_rel_l2_pct (lower is better)
- **Current best:** 4.619% (PR #2691, 4L/512d, AdamW lr=5e-4, T_max=30, no gc)
- **External target:** 3.71% (1.24x gap remaining)
- **Key context:** Lower LR (3e-4) failed on 3 independent experiments. Higher LR direction is the motivated next step.
- **Reproduce baseline:**
```bash
cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT chrome:

## Results

All 4 runs completed. **None beat the baseline.** Higher LR + gc=0.5 is strictly worse than the baseline lr=5e-4 (no gc) on DrivAerML — all configurations showed late-stage gradient instability, with lr=7e-4 and lr=1e-3 fully diverging.

### Summary Table

| Run | Config | Best Val (%) | Best Epoch | Final Test (%) | Diverged? |
|-----|--------|-------------|------------|----------------|-----------|
| 1 | lr=7e-4 + gc=0.5 | **11.13** | 75 | 62.87 | Yes (epoch ~80, grad_norm → ∞) |
| 2 | lr=8e-4 + gc=0.5 | **8.00** | 84 | 14.13 | Partial (epoch ~88, grad spikes) |
| 3 | lr=1e-3 + gc=0.5 | **12.38** | 63 | 69.93 | Yes (epoch ~70, grad_norm → ∞) |
| 4 | lr=5e-4 + gc=0.5 + WD=1e-2 | **8.26** | 91 | 12.44 | No (stable, but slow) |

**Baseline: 4.619%** (lr=5e-4, no gc, no extra WD)

### Key Observations

1. **gc=0.5 hurts DrivAerML at every LR tested.** The best result (Run 2, 8.00%) is 1.73x worse than baseline. Unlike AirfRANS where gc=0.5 yielded 28.9% improvement, DrivAerML's gradient landscape appears fundamentally different.

2. **Higher LR causes catastrophic instability.** lr=7e-4 and lr=1e-3 both diverged to grad_norm → Infinity. The cosine-t-max=30 schedule creates periodic LR peaks that, at higher base LRs, exceed a stability threshold around epoch 75-80.

3. **Run 4 (lr=5e-4 + gc=0.5 + WD=1e-2) was the most stable** but still ~1.79x worse than baseline. Adding weight decay prevented divergence but didn't help convergence — the compound regularization (gc + WD) is too conservative for this dataset.

4. **DrivAerML vs AirfRANS gradient dynamics:** AirfRANS benefits from gc because it tames high-variance surface gradients in the RANS field. DrivAerML's automotive surface meshes apparently have lower gradient variance at baseline, so gc=0.5 artificially constrains the optimization without providing stability gains.

### W&B Runs
- Run 1 (lr=7e-4+gc0.5): [vn91owo9](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/vn91owo9)
- Run 2 (lr=8e-4+gc0.5): [rwhi6g36](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/rwhi6g36)
- Run 3 (lr=1e-3+gc0.5): [o0fti0dg](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/o0fti0dg)
- Run 4 (lr=5e-4+gc0.5+WD1e-2): [dhsbltox](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/dhsbltox)

### Commands Used
```bash
# Run 1
CUDA_VISIBLE_DEVICES=0,1 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py --dataset drivaerml --optimizer adamw --lr 7e-4 --cosine-t-max 30 --grad-clip 0.5 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --wandb-group drivaerml-lr-gc --agent chrome --wandb-name "chrome/driv-lr7e4-gc05"

# Run 2
CUDA_VISIBLE_DEVICES=2,3 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py --dataset drivaerml --optimizer adamw --lr 8e-4 --cosine-t-max 30 --grad-clip 0.5 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --wandb-group drivaerml-lr-gc --agent chrome --wandb-name "chrome/driv-lr8e4-gc05"

# Run 3
CUDA_VISIBLE_DEVICES=4,5 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py --dataset drivaerml --optimizer adamw --lr 1e-3 --cosine-t-max 30 --grad-clip 0.5 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --wandb-group drivaerml-lr-gc --agent chrome --wandb-name "chrome/driv-lr1e3-gc05"

# Run 4
CUDA_VISIBLE_DEVICES=6,7 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --grad-clip 0.5 --weight-decay 1e-2 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --wandb-group drivaerml-lr-gc --agent chrome --wandb-name "chrome/driv-lr5e4-gc05-wd1e2"
```

### Suggested Follow-ups

1. **Try lr=6e-4 WITHOUT gc** — the sweet spot may be slightly above baseline LR but without gradient clipping, since gc appears harmful for DrivAerML.
2. **Warmup + higher LR** — a proper warmup phase (e.g. 5 epochs linear warmup to lr=7e-4) might avoid the early instability that gc was meant to prevent, without the convergence penalty.
3. **Longer cosine-t-max (60+)** — the T_max=30 schedule creates aggressive LR cycling that compounds with higher base LR. A gentler schedule might unlock higher LR without divergence.
4. **Dataset-specific gc tuning** — try gc=2.0 or gc=5.0 (much softer clip) to see if mild clipping helps DrivAerML without over-constraining gradients.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2827 because ADVISOR: Closing — gc=0.5 is definitively ruled out for DrivAerML. All 4 variants failed badly (best 8.00% vs 4.619% baseline). Higher LR + gc causes catastrophic divergence. Key insight: DrivAerML's narrow optimization basin cannot tolerate gc perturbations, unlike AirfRANS. Your result was clean and well-documented — important negative finding. You'll be reassigned to test a different DrivAerML direction, chrome..
